UI/Panels/Output/StepperPanel.cs

```diff
@@ -84,40 +81,72 @@ public void SetAdresses(List<ListItem> pins)
         {
-            if (OnManualCalibrationTriggered != null)
-            {
-                ManualCalibrationTriggeredEventArgs eventArgs = new ManualCalibrationTriggeredEventArgs();
-                eventArgs.StepperAddress = stepperAddressesComboBox.SelectedValue.ToString();
-                eventArgs.Steps = StepValues[trackBar1.Value];
-
-                OnManualCalibrationTriggered(this, eventArgs);
-            }
+            TriggerManualCalibration();
         }
 
+        private void TriggerManualCalibration()
+        {
+            if (OnManualCalibrationTriggered == null)
+                return;
+            
+            var eventArgs = new ManualCalibrationTriggeredEventArgs
+                {
+                    StepperAddress = stepperAddressesComboBox.SelectedValue.ToString(),
+                    Steps = stepValues[trackBar1.Value]
+                };
+
+             OnManualCalibrationTriggered(this, eventArgs);
+             button2.Enabled=true;
+            
+        }
+        
         private void button2_Click(object sender, EventArgs e)
         {
-            OnSetZeroTriggered?.Invoke(stepperAddressesComboBox.SelectedValue,
-                                   new StepperConfigChangedEventArgs()
-                                   {
-                                       StepperAddress = stepperAddressesComboBox.SelectedValue.ToString()
-                                   });
+            SetZero();
         }
 
+        private void SetZero()
+        {
+            OnSetZeroTriggered?.Invoke(stepperAddressesComboBox.SelectedValue,
+                new StepperConfigChangedEventArgs()
+                {
+                    StepperAddress = stepperAddressesComboBox.SelectedValue.ToString()
+                });
+            button2.Enabled=false;
+        }
+        
+        private void enter_KeyPress(object sender, KeyPressEventArgs e)
+        {
+            switch (e.KeyChar)
+            {
+                case (char)Keys.Return:
+                case (char)Keys.D0:
+                case (char)Keys.NumPad0:
+                    SetZero();
+                    break;
+                
+                case (char)Keys.Space:
+                    TriggerManualCalibration();
+                    break;
+            }
+        }
+        
+        
         private void inputRevTextBox_Validating(object sender, CancelEventArgs e)
         {
-            if (!(sender as Control).Parent.Enabled) return;
+            if (!((Control)sender).Parent.Enabled) return;
 
-            String value = (sender as TextBox).Text.Trim();
+            var value = ((TextBox)sender).Text.Trim();
 
             if (value == "") e.Cancel = true;
             if (e.Cancel)
             {
-                displayError(sender as Control, i18n._tr("uiMessagePanelsStepperInputRevolutionsMustNonEmpty"));
+                DisplayError((Control)sender, i18n._tr("uiMessagePanelsStepperInputRevolutionsMustNonEmpty"));
```

Comment:
Same.

Your code changes also are no changes. Please don't introduce such changes if we don't have a real benefit.

Reply:
Yes, I understand.